internals/secrethub/generate.go

```diff
@@ -13,16 +15,21 @@ var (
 	errGenerate = errio.Namespace("generate")
 
 	// ErrInvalidRandLength is returned when an invalid length is given.
-	ErrInvalidRandLength = errGenerate.Code("invalid_rand_length").Error("The secret length must be larger than 0")
+	ErrInvalidRandLength         = errGenerate.Code("invalid_rand_length").Error("The secret length must be larger than 0")
+	ErrCannotUseLengthArgAndFlag = errGenerate.Code("length_arg_and_flag").Error("length cannot be provided as an argument and a flag at the same time")
```

Comment:
Would we want to be more accepting and have the `--length` flag take precedence instead of erroring?

Reply:
I'm not sure whether that's better or whether that leads to unexpected behavior?

Consider e.g. this usage:
```
SECRETHUB_GENERATE_LENGTH=12
...
secrethub generate rand path/to/secret 30
```

When the environment variable is set at another place than the argument, it might not be clear that the argument is overridden.
In all other cases, values set as a flag/arg take precedence over environment variables. So one might expect that just looking at the command (`secrethub generate rand path/to/secret 30`) is enough.

Reply:
Ah yes, that is confusing and unpredictable indeed. Let's stick with erroring for now then. We can always be more forgiving later and we're going to drop this stuff eventually anyway.